### PR TITLE
feat: disable mysql server on datande when running standalone mode

### LINKED ISSUE #556: Remove MySQL server in datanode
### What type of enhancement is this?

Configuration

### What does the enhancement do?

In standalone mode, there is no need to start MySQL server in both frontend and datanode. We can check running mode in `datanode::server::Services` and start MySQL server only in distribuited mode.

### Implementation challenges

NA

### PR DESCRIPTION
**Pull request by v0y4g3r**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
Disable mysql server on datande when running standalone mode

## Checklist

- [X]  I have written the necessary rustdoc comments.
- [X]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
Fixes #556

### CODE REVIEW

**Review by MichaelScofield — APPROVED** [review]

File: src/datanode/src/server.rs [hunk 1]
```diff
@@ -45,33 +46,46 @@ impl Services {
...
         );
 
-        let mysql_io_runtime = Arc::new(
-            RuntimeBuilder::default()
-                .worker_threads(opts.mysql_runtime_size as usize)
-                .thread_name("mysql-io-handlers")
-                .build()
-                .context(RuntimeResourceSnafu)?,
-        );
+        let mysql_server = if let Mode::Distributed = opts.mode {
```
Comment (killme2008) [inline src/datanode/src/server.rs:49]: Using `match` would be better, considering `Mode` may have other variants.

File: src/datanode/src/server.rs [hunk 2]
```diff
@@ -17,11 +17,12 @@ use std::net::SocketAddr;
 use std::sync::Arc;
 
 use common_runtime::Builder as RuntimeBuilder;
+use common_telemetry::tracing::log::info;
+use frontend::frontend::Mode;
```
Comment (killme2008) [inline src/datanode/src/server.rs:21]: Move `Mode` to the server crate, it's not only used in the frontend. Not related to this PR, but recommend refactoring it.

**Review by killme2008 — COMMENTED** [review]
Good job, but we have to remove the MySQL options in `StandaloneOptions` too. https://github.com/GreptimeTeam/greptimedb/blob/develop/src/cmd/src/standalone.rs#L73-L74

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**